introspector/src/telemetry/mod.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of polkadot-introspector.
+//
+// polkadot-introspector is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// polkadot-introspector is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with polkadot-introspector.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::time::Duration;
+
+use clap::Parser;
+use priority_channel::{Receiver, TryRecvError};
+use subxt::utils::H256;
+use tokio::sync::broadcast::Sender as BroadcastSender;
+
+use crate::core::{TelemetryEvent, TelemetrySubscription, MAX_MSG_QUEUE_SIZE};
+
+#[derive(Clone, Debug, Parser)]
+#[clap(rename_all = "kebab-case")]
+pub(crate) struct TelemetryOptions {
+	/// Web-Socket URL of a telemetry backend
+	#[clap(name = "ws", long)]
+	pub url: String,
+	// Chain's genesis hash
+	#[clap(name = "chain", long)]
+	pub chain_hash: H256,
```

Comment:
Why do we need this ?

Reply:
Telemetry in the beginning sends events only about added chains. We need to subscribe to one of them to start getting its events.

Reply:
Ah, this looks like usability trouble. In practice do we have multiple chains on the same backend ?